slatedb/src/db/builder.rs

```diff
@@ -201,9 +203,9 @@ impl<P: Into<Path>> DbBuilder<P> {
         self
     }
 
-    /// Sets the block cache to use for the database.
-    pub fn with_block_cache(mut self, block_cache: Arc<dyn DbCache>) -> Self {
-        self.block_cache = Some(block_cache);
+    /// Sets the memory cache to use for the database.
```

Comment:
I'd probably expand this doc since it's more abstract now.

```suggestion
    /// Sets the memory cache to use for the database.
    ///
    /// SlateDB uses a cache to efficiently store and retrieve blocks and SST metadata locally.
    /// [`slatedb::db_cache::SplitCache`] is used by default.
```